### PR TITLE
Add search bar

### PR DESCRIPTION
**Pull request by alexsenden**
### Changes:
- Added route to get users and artists/albums/songs by partial name in the api
- Added a search bar modal that opens from a button in the appbar/mobile menu
- Search bar autocompletes
- Search bar and new post topic options say (artist), (album), etc. (we should still improve this though)
- Selecting a search result takes you to that result's individual page (artist/album/song pages are just temp pages for not until #27 #28 #29)

Screenshot 2023-11-06 221947

### CODE REVIEW

File: harmony-fe/src/components/autocomplete-li/index.tsx
Comment (Dankgen) [inline harmony-fe/src/components/autocomplete-li/index.tsx:1]: Empty? No export?

File: harmony-fe/src/components/search-bar/index.tsx
Comment (Dankgen) [inline harmony-fe/src/components/search-bar/index.tsx:1]: Same thing here

**Review by Dankgen — APPROVED** [review]
Forgot to approve. I just had those 2 little nitpicks

**Comment by alexsenden** [comment]
#22 #23 #24 #25

**Pull request merged**